src/Integration.Vsix/Suppression/SuppressionHandler.cs

```diff
@@ -0,0 +1,83 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using SonarLint.VisualStudio.Integration.Suppression;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.Suppression
+{
+    internal class SuppressionHandler
+    {
+        private readonly ILiveIssueFactory liveIssueFactory;
+        private readonly ISonarQubeIssuesProvider serverIssuesProvider;
+
+        public SuppressionHandler(ILiveIssueFactory liveIssueFactory, ISonarQubeIssuesProvider serverIssuesProvider)
+        {
+            if (liveIssueFactory == null)
+            {
+                throw new ArgumentNullException(nameof(liveIssueFactory));
+            }
+            if (serverIssuesProvider == null)
+            {
+                throw new ArgumentNullException(nameof(serverIssuesProvider));
+            }
+
+            this.liveIssueFactory = liveIssueFactory;
+            this.serverIssuesProvider = serverIssuesProvider;
+        }
+
+        public bool ShouldIssueBeReported(SyntaxTree syntaxTree, Diagnostic diagnostic)
```

Comment:
The logic in this method hasn't changed - it's just been moved from the _SuppressionManager_ class.